sources/packages/backend/libs/sims-db/src/entities/application-offering-change-requests.model.ts

```diff
@@ -0,0 +1,162 @@
+import {
+  Column,
+  Entity,
+  JoinColumn,
+  ManyToOne,
+  OneToOne,
+  PrimaryColumn,
+  RelationId,
+} from "typeorm";
+import { ColumnNames, TableNames } from "../constant";
+import { RecordDataModel } from "./record.model";
+import { Application } from "./application.model";
+import { EducationProgramOffering } from "./education-program-offering.model";
+import { User } from "./user.model";
+import { Note } from "./note.model";
+import { ApplicationOfferingChangeRequestStatus } from "./application-offering-change-request-status.type";
+
+const REASON_MAX_LENGTH = 500;
+/**
+ * Represents the list of application specific offering change request,
+ * which is requested by institution.
+ */
+@Entity({ name: TableNames.ApplicationOfferingChangeRequests })
+export class ApplicationOfferingChangeRequest extends RecordDataModel {
+  /**
+   * Auto-generated sequential primary key column.
+   */
+
+  @PrimaryColumn({
+    name: "id",
+    nullable: false,
+  })
+  id: number;
+
+  /**
+   * Student Application id related to application specific offering change request.
+   */
+  @RelationId(
```

Comment:
I see that adding `@RelationId` is to get the object's id without doing the join but what's the advantage of this instead of just adding `@Column`? @sh16011993 and I were trying to do a query using `findOne()` selecting a property decorated with `@RelationId` and it wasn't possible. We did some tests and checked selecting is just for columns. Then we changed it to `@Column` as test and everything seemed to work.

Reply:
Apparently `@RelationId` is causing more questions than benefits. I would be ok to stop using them.

Reply:
Removed